rules/prefer-t-regex.js

```diff
@@ -13,75 +13,183 @@ const create = context => {
 		'falsy'
 	]);
 
+	const equalityTests = new Set([
+		'is',
+		'deepEqual'
+	]);
+
+	// Find the latest reference to the given identifier's name.
 	const findReference = name => {
 		const reference = context.getScope().references.find(reference => reference.identifier.name === name);
-		const definitions = reference.resolved.defs;
-		return definitions[definitions.length - 1].node;
-	};
 
-	return ava.merge({
-		CallExpression: visitIf([
-			ava.isInTestFile,
-			ava.isInTestNode
-		])(node => {
-			// Call a boolean assertion, for example, `t.true`, `t.false`, …
-			const isBooleanAssertion = node.callee.type === 'MemberExpression' &&
-				booleanTests.has(node.callee.property.name) &&
-				util.getNameOfRootNodeObject(node.callee) === 't';
+		if (reference && reference.resolved) {
+			const definitions = reference.resolved.defs;
 
-			if (!isBooleanAssertion) {
-				return;
+			if (definitions.length === 0) {
+				return null;
```

Comment:
```suggestion
				return;
```

I prefer not using `null` at all.

Reply:
Ah, I didn't know you have that preference. I've removed that now.